### PR TITLE
Solves #2803: Implement applications page under Settings

### PR DESCRIPTION
**Pull request by gaeun**
This fixes issue #2803.
I added Instagram and Google connections under the Profile page, and implemented an Applications page under the Settings page. Below are screenshots of the changes I made:
screen shot 2017-01-04 at 9 28 50 pm
screen shot 2017-01-04 at 9 28 31 pm

### CODE REVIEW

**Comment by mariobehling** [comment]
Thanks for working on this.

In the issue it is stated: "Please create a tab "Applications" in the settings list at https://eventyay.com/settings/ similar to twitter"

Not in the profile section.

The apps should work similarly to the twitter apps page. Could you have a look at it please and compare it to your current implementation?

**Comment by gaeun** [comment]
@mariobehling I also created an Applications page in the settings list (the first screenshot). Here's a more comprehensive view:
screen shot 2017-01-04 at 9 42 11 pm
It shows connected applications as "Connected" -- isn't this what the issue asked for?

**Comment by mariobehling** [comment]
Ahh ok. Very good. 

Cause you wrote:

> I added Instagram and Google connections under the Profile page.

Let's not have the apps connections in the profile anymore. These connections should not be public.

At a later step the idea is to make the Profile into a public speaker profile, which will be something different.  However the details are not important at this point. Simply take it out of the profile, please. Thank you.

**Comment by gaeun** [comment]
Done! :)

**Comment by mariobehling** [comment]
Thank you. Travis is failing. Please check.

**Comment by gaeun** [comment]
Will get to this today, I got busy with school :p

File: app/views/users/profile.py [hunk 1]
```diff
@@ -63,6 +63,11 @@ def connect_instagram():
     instagram_auth_url, state = instagram.authorization_url(InstagramOAuth.get_auth_uri(), access_type='offline')
     return redirect(instagram_auth_url)
 
+@profile.route('/google_connect', methods=('GET', 'POST'))
```
Comment (niranjan94) [inline app/views/users/profile.py]: @gaeun we already have a callback for google at https://github.com/fossasia/open-event-orga-server/blob/development/app/views/utils_routes.py#L24 . Please reuse the same callback.

**Comment by gaeun** [comment]
@niranjan94 I removed the google authentication code -- we don't need it anyway, now that the social media connections are removed from the Profile.
@mariobehling Travis is failing due to a robot test, but when I run the same robot/tests on my localhost, all of them pass. I'm not sure what the problem is, since I ran the same scripts...
screen shot 2017-01-06 at 11 32 27 pm

**Comment by niranjan94** [comment]
@gaeun let me check this locally

**Comment by niranjan94** [comment]
@gaeun I can confirm with travis.

```
/usr/bin/python2.7 /usr/local/bin/robot tests/robot
==============================================================================
Robot                                                                         
==============================================================================
Robot.Login :: A test suite with a single test for valid login.               
==============================================================================
Valid Login                                                           | FAIL |
Location should have been 'http://localhost:5000/' but was 'http://localhost:5000/login/'
------------------------------------------------------------------------------
Invalid Login nonexistent user                                        | PASS |
------------------------------------------------------------------------------
Invalid Login wrong password                                          | FAIL |
Page should have contained text 'Incorrect Password' but did not
------------------------------------------------------------------------------
Robot.Login :: A test suite with a single test for valid login.       | FAIL |
3 critical tests, 1 passed, 2 failed
3 tests total, 1 passed, 2 failed
==============================================================================
Robot                                                                 | FAIL |
3 critical tests, 1 passed, 2 failed
3 tests total, 1 passed, 2 failed
==============================================================================
Output:  /home/niranjan94/Projects/open-event-orga-server/output.xml
Log:     /home/niranjan94/Projects/open-event-orga-server/log.html
Report:  /home/niranjan94/Projects/open-event-orga-server/report.html

Process finished with exit code 2
```

Robot tests are failing for me too

**Comment by gaeun** [comment]
@niranjan94 Can you help me identify why this is happening? I've tried fixing it for hours but I can't get it to work on travis.

**Comment by niranjan94** [comment]
@gaeun your migration files
You seem to have duplicates

**Comment by niranjan94** [comment]
Hold on ... Migrations are a bit tricky ...

**Comment by niranjan94** [comment]
@gaeun let's see what happens now ...

**Comment by gaeun** [comment]
@niranjan94 Hmm it seems like Travis is still failing :/

**Comment by gaeun** [comment]
@niranjan94 Travis passed! 😮 Thank you so much!

**Comment by mariobehling** [comment]
Good job both! Thank you.

**Comment by mariobehling** [comment]
Unfortunately tests on open-event-dev site show issue is not solved.
* Facebook does not show up
* Apps show connected without info on name as shown in screenshot
* Apps show connected, but are not connected
* Instagram misses app keys, but still shows as connected

screenshot from 2017-01-07 05-13-16

**Comment by gaeun** [comment]
@mariobehling One question: now that the Profile page is gone, where can users input their social media information?

**Comment by mariobehling** [comment]
The profile page is still there, but not so important, because this is not a social network at the moment.

The idea is that we collect data mainly for speaker and participant profiles. So, we save profiles of speakers on the profile page and speaker could choose which profile shows up at which event. Meetup.com has something similar, but this is not so straight forward and we need to think this through first. I think for the Beta version of the app this is not so important.

**Comment by gaeun** [comment]
@mariobehling For now, how should I collect the information that's necessary to display information on the Applications page?

**Comment by mariobehling** [comment]
Please try to connect a twitter account with a Facebook account. Then implement it in the same way in Open Event, instead of twitter - it is Open Event. I think it is easiest if we just follow other services. The connection we make here, should enable people to login with their other social media accounts if they like.

**Comment by gaeun** [comment]
I'm working on making a few changes, and so far I have:
- Made Facebook show up
- Allowed users to connect instead of displaying "Connected"
- Made it so that info doesn't show up if not connected

screen shot 2017-01-08 at 11 47 05 am

There is one problem though -- I can't test OAuth in my local environment, and whenever I try to connect the apps I get errors.

**Comment by niranjan94** [comment]
@gaeun have you added the proper keys to your app settings ?

**Comment by gaeun** [comment]
@niranjan94 How do you do that?

**Comment by niranjan94** [comment]
@gaeun at `/admin/settings/`

**Comment by gaeun** [comment]
@niranjan94 /admin/settings in the source code, or on the website? I can't seem to find it in the code... OAuth is confusing :p

**Comment by niranjan94** [comment]
On the website ... @gaeun please go through the entire application ... Click on all the links you can find ... See where they lead you to ... explore

**Comment by gaeun** [comment]
@niranjan94 I've searched everywhere on the website but I can't find an admin page, even after I created an event. Do you mean the personal Settings page? If you can guide me to where I can set up the app keys I'd like to work on fixing the problem.

**Comment by mariobehling** [comment]
@gaeun Admin is only accessible on your own installation with the account you set up the application.

screenshot from 2017-01-10 03-58-27

**Comment by gaeun** [comment]
@mariobehling Ah I see, thank you!

**Pull request merged**